### DISCUSSION TITLE
What's the best way to configure reusable ops

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8548] [Q&A]
Okay I think I got an easy one for y'all

I was looking over [Using the Same Op Twice](https://docs.dagster.io/concepts/ops-jobs-graphs/jobs-graphs#using-the-same-op-twice) because I want to have an op that just runs a SQL query and returns the result, but I want to be able to configure which query it runs. How would I go about setting that up in the config? Can I use the alias method and then reference that alias in the op config?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1648749623830799?thread_ts=1648749623.830799&cid=C01U954MEER).*

**Comment by yuhan** [comment]
You can set up your op with a config schema like:
```python
@op(config_schema={"sql_query": str})
def run_query(context):
    return somehow_run_query(context.op_config["sql_query"])
```

Then, you can create versions of this run_query op with this config "baked in" by doing stuff using [`configured`](https://docs.dagster.io/concepts/configuration/configured#configured-api) like: 
```python
select_1_op = run_query.configured({"sql_query": "select 1;"}, name="select_1")
```

**Reply by javiermas** [reply]
Now that the configured API is deprecated, I don't really see an equivalent way to create re-usable ops with the new config methods. Any thoughts here?

**Reply by dmullen17** [reply]
I'm interested in this answer too!

**Reply by dmullen17** [reply]
This solution uses aliases!
```
from dagster import op, job, ScheduleDefinition, Config, RunConfig, asset

class printTextConfig(Config):
    text: str

@op
def printText(config: printTextConfig) -> str:
        print(config.text)
        return(config.text)

# Define the default configuration for each op instance
default_config = RunConfig(
    ops={
        'sayHi': {"config": {"text": 'hi'}},
        'sayHello': {"config": {"text": 'hello'}}
    }
)

@job(config=default_config)
def sayGreetings():
    hi = printText.alias('sayHi')()
    hello = printText.alias('sayHello')()
```
